src/core/qgsnetworkaccessmanager.cpp

```diff
@@ -203,11 +204,19 @@ QNetworkReply *QgsNetworkAccessManager::createRequest( QNetworkAccessManager::Op
   }
 #endif
 
-  emit requestAboutToBeCreated( QgsNetworkRequestParameters( op, req ) );
+  static QAtomicInt sRequestId = 0;
```

Comment:
Are you counting on the fact that older requests timed out when this eventually overflows?

Reply:
The server instance could be up for quite some time and in case of cascading WMS it would make a lot of requests.

Reply:
Fair enough, but in that case the id will just "wrap around" without harm. It's only used for debugging any linking a request with a response.